### PR TITLE
DAOS-9044 event: if async op failed, we are not resetting the priv event

### PR DESCRIPTION
**Pull request by mchaarawi**
Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/api/event.c [hunk 1]
```diff
@@ -1261,18 +1264,24 @@ daos_event_priv_wait()
 		/** progress succeeded, loop can exit if event completed */
 		if (rc == 0) {
 			rc = ev_thpriv.ev_error;
-			if (rc)
+			if (rc) {
+				rc2 = daos_event_priv_reset();
```
Comment (liuxuezhao) [inline src/client/api/event.c:1268]: why reset it here?
the "ev_thpriv.ev_error != 0" just means the async operation failed right? reset will lost the fail code, no?

Comment (mchaarawi) [reply src/client/api/event.c:1268]: the error itself is saved first returned from the function itself, so it won't get lost?
the purpose of this is a strange bug i saw in :
https://github.com/daos-stack/daos/pull/7177
where a DAOS operation was failing (as expected) but the error code was getting saved and returned in a subsequent operation (that has passed, but still reported the error at the API level from the previous operation).

Comment (liuxuezhao) [reply src/client/api/event.c:1268]: ok, maybe reset ev_thpriv.ev_error = 0 is enough, but daos_event_priv_reset() seems no bad effect, just seems a little overkill.

**Review by liuxuezhao — APPROVED** [review]

File: src/client/api/event.c (hunk 1 above)
Comment (mchaarawi) [reply src/client/api/event.c:1268]: i see. i can try that in the other PR after this lands and i merge there.

File: src/client/api/event.c [hunk 2]
```diff
@@ -406,8 +406,7 @@ daos_event_launch(struct daos_event *ev)
 	if (daos_handle_is_valid(evx->evx_eqh)) {
 		eqx = daos_eq_lookup(evx->evx_eqh);
 		if (eqx == NULL) {
-			D_ERROR("Can't find eq from handle %"PRIu64"\n",
-				evx->evx_eqh.cookie);
+			D_ERROR("Can't find eq from handle %"PRIu64"\n", evx->evx_eqh.cookie);
```
Comment (jolivier23) [inline src/client/api/event.c:409]: minor

s/%"PRIu64"/"DF_U64"

**Review by jolivier23 — APPROVED** [review]

File: src/client/api/event.c (hunk 2 above)
Comment (mchaarawi) [reply src/client/api/event.c:409]: i didn't change anything there other than move the argument to the same line due to the increase in width ;-)

**Pull request merged**